### DISCUSSION TITLE
How to correctly link to a root file (like javascript file)

### DISCUSSION THREAD

**Discussion by Vinc26** [GetPublii/Publii#1751] [Help] [answered]
Hi, I simply want to link a javascript file that works for any page of my website.

So I've first uploaded the js file using file manager, to the root directory.
Then, I've added this kind of script in the head custom html : `<script type='text/javascript' src="/jquery-3.7.1.min.js"></script>`

But it doesn't work offline (preview locally), neither online.

If I don't use the `/` at the beginning of the file, it works only in the homepage.

**Comment by bjazmoore** [comment]
Move your javascript file to the /assets/js/.. folder of the theme.  Use the following to reference the file:
[https://getpublii.com/dev/js-helper/](https://getpublii.com/dev/js-helper/)
<script src="{{js "jquery-3.7.1.min.js"}}"></script>

**Reply by candidexmedia** [reply]
Just a note: `<script src="{{js "jquery-3.7.1.min.js"}}"></script> ` won't work if you add it to the Head section of Custom HTML. It'll need to be loaded in the site's theme files (I believe the file is called `head.hbs`, in the partials folder).

Alternately @Vinc26 you can load [jquery via the CDN hosted link](https://releases.jquery.com/) if you don't want to touch theme files, and only want to edit the Custom HTML in the GUI.

**Reply by bjazmoore** [reply]
Yes - I did not mean to imply that this setting could be used outside of modifying the Handlebars theme file.

**Comment by Vinc26** [comment]
Thanks :) I'm starting to understand that I'm going to have to overrides the theme. What annoys me because every time the theme is going to be updated, I will have to resume each of my changes.
And I don't want to use CDN host link because of GDPR.

**Reply by candidexmedia** [reply, marked as answer]
[Theme overrides](https://getpublii.com/dev/theme-overrides/) safeguard from update overrides.

> And I don't want to use CDN host link because of GDPR.

I don't know which server you use, but is there a possibillity of hosting them yourself on a separate directory, and linking to those files in the GUI?

**Reply by Vinc26** [reply]
Yes !!! Thanks for this idea ! I'll try.
